### ISSUE TITLE
Migrate the Enter and ShiftEnter plugins to beforeInput

### ISSUE THREAD

**Issue by oleq** [ckeditor/ckeditor5#7999]
## Provide a description of the task

At this moment `EnterObserver` is used to detect <kbd>Enter</kbd> and <kbd>Shift+Enter</kbd> keystrokes. This can be handled via `beforeInput` `insertParagraph` and `insertLineBreak` event types, though, for instance:

```javascript
editor.editing.view.document.on( 'beforeinput', ( evt, data ) => {
	const domEvent = data.domEvent;

	if ( domEvent.inputType === 'insertLineBreak' ) {
		editor.execute( 'shiftEnter' );

		evt.stop();
		data.preventDefault();
	}
} );
```

**Comment by oleq** [comment]
Features that listen to `enter`:
* `BlockQuoteEditing`,
* `CodeBlockEditing`,
* `ListEditing`,
* `WidgetTypeAround`

**Comment by oleq** [comment]
Idea: Split the logic inside `EnterObserver` to either use `keydown` or `beforeInput`.

**Comment by oleq** [comment]
Closed in [#817](https://github.com/ckeditor/ckeditor5/pull/8176)6.
